### PR TITLE
Feature/ussd nurse rapidpro

### PR DESCRIPTION
**Pull request by OMosimege**
Moving NurseConnect over to rapid pro

All Identity Store functions will be replaced with the NurseConnect Companion App functions

Comments are added for functions that will be added later on

### CODE REVIEW

File: test/ussd_nurse_rapidpro.test.js [hunk 1]
```diff
@@ -210,16 +210,35 @@ describe('app', function() {
...
                 describe('user enters msisdn', function(){
+
+                  //for msisdn that has not opted out
+                  it('should go to state_faccode', function(){
+                      return tester
+                      .setup.user.addr('27820001003')
+                      .inputs(
+                          {session_event: 'new'}
+                          ,'1' //chooses to start nurse connect
+                          ,'1'
```
Comment (miltontony) [inline test/ussd_nurse_rapidpro.test.js]: Please add comment on what this input is (similar to other test cases)

File: test/ussd_nurse_rapidpro.test.js [hunk 2]
```diff
@@ -210,16 +210,35 @@ describe('app', function() {
...
-                        .setup.user.addr('0000000000') //should be 27820001004 in fixtures
+                        .setup.user.addr('27820001004') //should be 27820001004 in fixtures
                         .inputs(
-                            {session_event: 'new'}  
+                            {session_event: 'new'}
                             ,'1' //chooses to start nurse connect
-                            ,'1'  // chooses to sign up
-                            ,'1' //chooses yes to subscribe
-                            ,'0000000000'  // state_msisdn
+                            ,'1'
```
Comment (miltontony) [inline test/ussd_nurse_rapidpro.test.js]: Please add comment on what this input is (similar to other test cases)

File: test/ussd_nurse_rapidpro.test.js [hunk 3]
```diff
@@ -249,14 +310,13 @@ describe('app', function() {
...
-                 
+
         }); //end of state_start
-    
-    
+
+
     });
 });
-
```
Comment (miltontony) [inline test/ussd_nurse_rapidpro.test.js:262]: Always ensure new line at the end of the file (there's usually a setting in your editor that automatically sets this for you)

**Review by miltontony — COMMENTED** [review]
Minor changes suggested.

I do feel though that your PR is not named/described properly. This feels like it's just about adding the "no subscription" state and the rest of the changes earmarked in the description will follow in later PRs.

File: test/ussd_nurse_rapidpro.test.js (hunk 3 above)
Comment (OMosimege) [reply test/ussd_nurse_rapidpro.test.js:262]: noted, will do

File: test/ussd_nurse_rapidpro.test.js (hunk 2 above)
Comment (OMosimege) [reply test/ussd_nurse_rapidpro.test.js]: thanks will do

File: test/ussd_nurse_rapidpro.test.js (hunk 1 above)
Comment (OMosimege) [reply test/ussd_nurse_rapidpro.test.js]: Thanks, will add those. And regarding the description, you're right, that description explains the branch, and has no description for the pull-request

**Comment by OMosimege** [comment]
This PR adds the state_no_subscription, and further tests that lead to state_no_subcription, or subsequently state_faccode

File: go-app-ussd_nurse_rapidpro.js [hunk 4]
```diff
@@ -150,23 +150,20 @@ go.app = function() {
...
                             //address will be set here
-                        };
-                        return is
-                        .optin(optin_info)
-                        .then(function() {
-                            return 'state_faccode';
-                        });
-                    }else {
-                        return self.states.create('state_no_subscription');
+                            //address type will be set here
```
Comment (sewagodimo) [inline go-app-ussd_nurse_rapidpro.js]: wouldn't block comments be better

File: src/ussd_nurse_rapidpro.js [hunk 5]
```diff
@@ -147,23 +147,20 @@ go.app = function() {
...
-                        return is
-                        .optin(optin_info)
-                        .then(function() {
-                            return 'state_faccode';
-                        });
-                    }else {
-                        return self.states.create('state_no_subscription');
+                            //address type will be set here
+                            //config name will be set here
+
```
Comment (sewagodimo) [inline src/ussd_nurse_rapidpro.js]: could also be block comments

**Review by sewagodimo — COMMENTED** [review]
comment about block comments,
I think Javascript uses the java /* */ block comments

File: go-app-ussd_nurse_rapidpro.js (hunk 4 above)
Comment (OMosimege) [reply go-app-ussd_nurse_rapidpro.js]: These all will be replaced once new functions that replace Seed functions, but will change thanks 👍

**Review by sewagodimo — APPROVED** [review]
:cake:

**Pull request merged**